### ISSUE TITLE
Do not do unsafe autocorrection for `Lint/UselessAssignment` 

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#13103]
The cop is already marked as unsafe with the example using `||=` provided and the problematic variable declaration. 
But, there is another example I just encountered.

```ruby
foo += bar ? 1 : 2 
```

Which is incorrectly gets autocorrected to `foo + bar ? 1 : 2` (which is equivalent to `(foo + bar) ? 1 : 2`, but should be autocorrected to something like `foo + (bar ? 1 : 2)` (note the operator precedences in both examples).

What if we stop autocorrecting for such cases (and for cases like mentioned in the "safety" doc-block)?

**Comment by koic** [comment]
I like the idea. It may be possible to modify it so that autocorrection is performed only in safe cases.
